packages/wallets/hyperpay/src/adapter.ts

```diff
@@ -0,0 +1,203 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    EventEmitter,
+    scopePollingDetectionStrategy,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+
+interface HyperPayWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface HyperPayWallet extends EventEmitter<HyperPayWalletEvents> {
+    isHyperPay?: boolean;
+    publicKey?: { toBytes(): Uint8Array };
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+}
+
+interface HyperPayWindow extends Window {
+    solana?: HyperPayWallet;
```

Comment:
Please change this to `window.hyperPay`. Using `window.solana` means that multiple wallets conflict for the same object, which makes detection impossible. New adapters that use `window.solana` aren't being accepted, and ones that currently use it have committed to a migration away from it.